Neos.Media/Classes/Domain/Model/AssetSource/Neos/NeosAssetProxyRepositoryInterface.php

```diff
@@ -0,0 +1,266 @@
+<?php
+namespace Neos\Media\Domain\Model\AssetSource\Neos;
+
+/*
+ * This file is part of the Neos.Media package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+  *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Media\Domain\Model\AssetSource\AssetNotFoundException;
+use Neos\Media\Domain\Model\AssetSource\AssetProxy\AssetProxyInterface;
+use Neos\Media\Domain\Model\AssetSource\AssetProxyQueryResultInterface;
+use Neos\Media\Domain\Model\AssetSource\AssetProxyRepositoryInterface;
+use Neos\Media\Domain\Model\AssetSource\AssetTypeFilter;
+use Neos\Media\Domain\Model\AssetSource\SupportsCollections;
+use Neos\Media\Domain\Model\AssetSource\SupportsSorting;
+use Neos\Media\Domain\Model\AssetSource\SupportsTagging;
+use Neos\Flow\Annotations\Inject;
+use Neos\Flow\ObjectManagement\ObjectManagerInterface;
+use Neos\Flow\Persistence\Exception\InvalidQueryException;
+use Neos\Flow\Persistence\QueryInterface;
+use Neos\Media\Domain\Model\AssetCollection;
+use Neos\Media\Domain\Model\AssetInterface;
+use Neos\Media\Domain\Model\Tag;
+use Neos\Media\Domain\Repository\AssetRepository;
+use Neos\Media\Domain\Repository\AudioRepository;
+use Neos\Media\Domain\Repository\DocumentRepository;
+use Neos\Media\Domain\Repository\ImageRepository;
+use Neos\Media\Domain\Repository\VideoRepository;
+
+final class NeosAssetProxyRepositoryInterface implements AssetProxyRepositoryInterface, SupportsSorting, SupportsCollections, SupportsTagging
```

Comment:
Was it really intended to be named `NeosAssetProxyRepositoryInterface`? Shouldn't that rather be `NeosAssetProxyRepository` or even `NeosAssetRepository` ?

Reply:
Mhm, no?

Reply:
No, that was of course not the intention ;-) And I fixed that in https://github.com/neos/neos-development-collection/pull/1979/commits/71138fcef84241647419c66ee0ec0fcee1bfe6d5 and https://github.com/neos/neos-development-collection/pull/1979/commits/f0b17f398357feb00a71067be81d714c929d45e5.